### PR TITLE
ESRI-239 - Redesign of Login Page

### PR DESCRIPTION
**Pull request by AleksaMilivojevicCycloMedia**
ESRI-239 - Redesign of Login Page

### CODE REVIEW

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs [hunk 1]
```diff
@@ -251,7 +257,7 @@ protected override Task CancelAsync()
     {
       _login.Username = _username;
       _login.Password = _password;
-      _login.IsOAuth = _isOAuth;
+      //_login.IsOAuth = _isOAuth;
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs]: Do not leave commented source code.

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs [hunk 2]
```diff
@@ -270,8 +276,9 @@ protected override void NotifyPropertyChanged([CallerMemberName] string property
     public void Save()
     {
       _login.Save();
-      // ReSharper disable once ExplicitCallerInfoArgument
       NotifyPropertyChanged("Credentials");
+      NotifyPropertyChanged("Username");
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs:267]: Double check why is this needed here.

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicAuthLoggedInConverter.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  class BasicAuthLoggedInConverter : IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicAuthLoggedInConverter.cs]: Name is unclear

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicAuthLoggedInConverter.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  {
+    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+      if (value is bool credentials && credentials)
+      {
+        return 0;
+      }
+      return 2;
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicAuthLoggedInConverter.cs]: Return values are unclear.

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicLoginConverter.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  public class BasicLoginConverter : IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicLoginConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicLoginRevertConverter.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  public class BasicLoginRevertConverter : IValueConverter
+  {
+    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/BasicLoginRevertConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/IsOAuthToRowConverter.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  
+
+  public class IsOAuthToRowConverter : IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/IsOAuthToRowConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/LoginBasicConverter.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  public class LoginBasicConverter : IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/LoginBasicConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/MarginSplitConverter.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  class MarginSplitConverter : IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/MarginSplitConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/OAuthUsernameDisplayConverter.cs [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+using static StreetSmartArcGISPro.Configuration.File.Login;
+
+namespace StreetSmartArcGISPro.AddIns.Views.Converters
+{
+  class OAuthUsernameDisplayConverter: IValueConverter
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/OAuthUsernameDisplayConverter.cs]: Unclear name

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/SignInEnabled.cs [hunk 11]
```diff
@@ -16,6 +16,7 @@
  * License along with this library.
  */
 
+using CefSharp.DevTools.CSS;
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Converters/SignInEnabled.cs]: Why do we need this.

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Login.xaml.cs [hunk 12]
```diff
@@ -1,5 +1,10 @@
-﻿using System.Windows;
+﻿using ArcGIS.Desktop.Internal.Mapping;
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Login.xaml.cs]: Is this needed?

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/Properties/Resources.fr.resx [hunk 13]
```diff
@@ -221,6 +221,9 @@ Type de produit:</value>
   </data>
   <data name="LoginCheck" xml:space="preserve">
     <value>Vérifier</value>
+  </data>
+	<data name="LogoutCheck" xml:space="preserve">
+    <value>Log out</value>
```
Comment (bervacanin-cy) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/Properties/Resources.fr.resx]: Double check if we can get all translations for fr.

**Review by bervacanin-cy — CHANGES_REQUESTED** [review]

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs [hunk 14]
```diff
@@ -182,11 +188,11 @@ public string Password
...
       set
       {
-          if (_login.Password != value)
-          {
-            IsModified = true;
-            _login.Password = value;
-            NotifyPropertyChanged();
+        if (_login.Password != value)
+        {
+          IsModified = true;
```
Comment (martin-sven-kovacic-valcon) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs:186]: why do we need this "IsModified" variable?

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/Configuration/File/Login.cs [hunk 15]
```diff
@@ -174,6 +174,7 @@ public bool Credentials
         {
           _credentials = value;
           OnPropertyChanged();
+          OnPropertyChanged("Credentials");
```
Comment (martin-sven-kovacic-valcon) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/Configuration/File/Login.cs]: duplication

**Review by martin-sven-kovacic-valcon — CHANGES_REQUESTED** [review]

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs [hunk 16]
```diff
@@ -127,11 +127,16 @@ private async Task SignOutOAuth()
         EventLog.Write(EventLog.EventType.Error, $"Street Smart: (Login.cs) (SignOutOAuth) {ex}");
       }
 
+      _login.IsOAuth = false;
+      _login.Credentials = false;
+      NotifyPropertyChanged("Credentials");
```
Comment (martin-sven-kovacic-valcon) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Pages/Login.cs:125]: duplication

**Review by martin-sven-kovacic-valcon — CHANGES_REQUESTED** [review]

File: client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Login.xaml [hunk 17]
```diff
@@ -47,43 +52,161 @@
...
-            </Grid.ColumnDefinitions>
-            <TextBlock Grid.Column="0" FontSize="12" Foreground="{DynamicResource Esri_Gray155}" TextAlignment="Right" util:LocalizationProvider.ContentId="LoginStatus" Width="100" Margin="0,5,5,0"/>
-            <TextBlock Grid.Column="1" FontSize="12" Foreground="{DynamicResource Esri_Gray155}" TextAlignment="Center" util:LocalizationProvider.ContentId="{Binding Credentials, Converter={StaticResource LoginStatusConverter}}" HorizontalAlignment="Stretch" Width="Auto" Margin="0,5,0,0" Visibility="{Binding IsOAuth, Converter={StaticResource InverseBoolToVisibility}}"/>
-            <TextBlock Grid.Column="1" FontSize="12" Foreground="{DynamicResource Esri_Gray155}" TextAlignment="Center" util:LocalizationProvider.ContentId="{Binding OAuthAuthenticationStatus, Converter={StaticResource OAuthSignStatus}}" HorizontalAlignment="Stretch" Width="Auto" Margin="0,5,0,0" Visibility="{Binding IsOAuth, Converter={StaticResource BoolToVisibility}}"/>
-            <fa:ImageAwesome Grid.Column="2" Icon="Spinner" Spin="True" Height="18" Width="18" Visibility="{Binding OAuthAuthenticationStatus, Converter={StaticResource SigningInProgressToVisibility}}" />
```
Comment (martin-sven-kovacic-valcon) [inline client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/Views/Login.xaml:81]: where did the spinner go?

**Review by martin-sven-kovacic-valcon — APPROVED** [review]

**Review by bervacanin-cy — APPROVED** [review]

**Pull request merged**